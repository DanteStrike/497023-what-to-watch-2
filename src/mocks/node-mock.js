export const createNodeMock = (element) => {
  if (element.type === `video`) {
    return {
      src: null,
      isMuted: null,
      poster: null,
      addEventListener: () => {}
    };
  }
  return null;
};
