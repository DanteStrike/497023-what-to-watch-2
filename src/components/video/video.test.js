import React from "react";
import renderer from "react-test-renderer";

import {createNodeMock} from "../../mocks/node-mock.js";
import Video from "./video.jsx";


it(`Render correctly VideoPlayer component`, () => {
  const VideoPlayerComponent = renderer
    .create(
        <Video
          isActivePlayer={false}
          src={`https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`}
          poster={`img/johnny-english.jpg`}
          isMuted={true}
        />,
        {
          createNodeMock
        }
    ).toJSON();

  expect(VideoPlayerComponent).toMatchSnapshot();
});
