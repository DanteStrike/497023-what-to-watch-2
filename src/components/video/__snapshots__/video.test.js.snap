// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly VideoPlayer component 1`] = `
<video
  style={
    Object {
      "backgroundColor": "black",
      "display": "block",
      "height": "100%",
      "left": "50%",
      "minHeight": "100%",
      "minWidth": "100%",
      "position": "absolute",
      "top": "50%",
      "transform": "translate(-50%, -50%)",
      "width": "100%",
    }
  }
/>
`;
